backend/api/serializers.py

```
@@ -13,21 +12,23 @@
-            "onboarding_completed",
+            "onboarding_completed", 
         ]
-
+        read_only_fields = ['onboarding_completed']
 
 class UserBasicSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
-        fields = ["id", "email", "first_name", "last_name"]
-
+        fields = ["id", "email", "first_name", "last_name", "role"]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)